python/lsst/atmospec/nightAnalysis.py

```diff
@@ -78,56 +73,61 @@ def getLineValue(table, lineName, columnName, nanForMissingValues=True):
         """Match object and check is dispersed"""
-        filt = self.butler.queryMetadata('raw', 'filter', dayObs=self.dayObs, seqNum=seqNum)[0]
+        where = "exposure.day_obs = dayObs and exposure.seq_num = seqNum"
+        records = list(self.butler.registry.queryDimensionRecords("exposure",
+                                                                  where=where,
+                                                                  bind={"dayObs": self.dayObs,
+                                                                        "seqNum": seqNum},
+                                                                  instrument=self.instrument,
+                                                                  ))
+        filt = records[0].physical_filter
         grating = filt.split(FILTER_DELIMITER)[1]
         if "ronchi" in grating:
             return True
         return False
 
-    def _expIdFromSeqNum(self, seqNum):
-        return self.butler.queryMetadata('raw', 'expId', dayObs=self.dayObs, seqNum=seqNum)[0]
-
-    def _getTableFilename(self, seqNum):
-        """Return the table filename, or None if not found for the dataId"""
-        expId = self._expIdFromSeqNum(seqNum)
-        try:
-            tablePath = self.butler.getUri('spectractorOutputRoot', expId=expId)
-        except butlerExceptions.NoResults:
-            return None
-
-        tableFilename = os.path.join(tablePath, 'extractedLines.fits')
-        return tableFilename
-
     def _readOneExtractionFile(self, seqNum):
-        filename = self._getTableFilename(seqNum)  # whole thing should really just be a butler.get()
-        if not filename:
+        datasetType = "extraction"
+        try:
+            return self.butler.get(datasetType,
+                                   seq_num=seqNum,
+                                   day_obs=self.dayObs,
+                                   instrument=self.instrument)
+        except LookupError:
             return None
 
-        if os.path.exists(filename):
-            table = Table.read(filename)
-            return table
-        return None
-
     def _loadExtractions(self, ignoreSeqNums):
-        allSeqNums = self.butler.queryMetadata('raw', 'seqNum', dayObs=self.dayObs, object=self.targetName)
-
+        # Get all the observations for the night.
+        where = "exposure.day_obs = dayObs and exposure.target_name = targetName"
+        records = self.butler.registry.queryDimensionRecords("exposure",
+                                                             where=where,
+                                                             instrument=self.instrument,
+                                                             bind={"dayObs": self.dayObs,
+                                                                   "targetName": self.targetName},
+                                                             )
+        allSeqNums = [r.seq_num for r in records]
```

Comment:
I think this should be `sorted([r.seq_num for r in set(records)])` to de-dupe, and because Gen2 returned them in order (and if it didn't then this was probably something I should have put in before).

Reply:
Okay. If you are always wanting them sorted I can do that. I was being careful elsewhere to retain the ordering in allSeqNums.

Reply:
Looks like you sort them a few lines below. We don't think `queryDimensionRecords` can return duplicates.

Reply:
OK, sounds good then, thanks.

Reply:
@TallJimbo tells me that he's not 100% sure that a future improvement to queryDataIds might not result in duplicates so I have added a set and sort below this line after things have been filtered out.